src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -1148,6 +1162,26 @@ public boolean hasWorkerOpenRequestsFiltered(@NotNull final ICitizenData citizen
         return getOpenRequests(citizen).stream().anyMatch(selectionPredicate);
     }
 
+    @Override
+    public boolean hasOpenSyncRequest(@NotNull final ICitizenData citizen)
+    {
+        final Map<Integer, Collection<IToken<?>>> requestIds = getOpenRequestsByCitizen();
+        if (!requestIds.containsKey(citizen.getId()))
```

Comment:
why not call the function hasWorkerOpenRequests for this part?

Reply:
ye thats simpler